src/application/utils/transaction.ts

```diff
@@ -119,16 +152,39 @@ export async function blindAndSignPset(
     throw new Error('blindPSET error: not fully blinded');
   }
 
-  const signedPset = await signPset(blindedPset, identities);
+  const signedPsetBase64 = await signPset(blindedPset, identities);
+
+  const pset = Psbt.fromBase64(signedPsetBase64);
+  if (!skipSigValidation) {
+    if (!pset.validateSignaturesOfAllInputs(sigValidator)) {
+      throw new Error('PSET is not fully signed');
+    }
+  }
 
-  const decodedPset = decodePset(signedPset);
-  if (!decodedPset.validateSignaturesOfAllInputs()) {
-    throw new Error('PSET is not fully signed');
+  for (let i = 0; i < pset.txInputs.length; i++) {
+    const input = pset.data.inputs[i];
+    // we need to use special finalizer in case of tapscript
+    if (atLeastOne(input.tapLeafScript) && atLeastOne(input.tapScriptSig)) {
```

Comment:
It can happen a tapscript branch does not have any signature tho: ie. a dumb `OP_ADD 3 OP_EQUAL` does not need any signature there, but needs two elements on the stack

Reply:
this "special finalizer" is mainly here for the "auto-spendable by marina" case. If the leaf does not need signature, it can't be finalized by marina (user needs to use a special finalizer)